packages/@cdktf/cli-core/src/lib/synth-stack.ts

```diff
@@ -153,21 +153,50 @@ Command output on stdout:
       process.exit(1);
     }
 
-    if (!(await fs.pathExists(path.join(outdir, Manifest.fileName)))) {
-      const errorMessage = `ERROR: synthesis failed, because app was expected to call 'synth()', but didn't. Thus "${outdir}/${Manifest.fileName}"  was not created.`;
+    // end performance timer
+    const endTime = performance.now();
+
+    let stacks: SynthesizedStack[] = [];
+    try {
+      stacks = await SynthStack.readSynthesizedStacks(outdir);
+    } catch (e: any) {
+      const errorMessage = `ERROR: synthesis failed, because app was expected to call 'synth()', but didn't. Thus "${outdir}/${Manifest.fileName}"  was not created: ${e}`;
       if (graceful) {
         throw new Error(errorMessage);
       }
       logger.error(errorMessage);
       process.exit(1);
     }
 
-    // end performance timer
-    const endTime = performance.now();
+    await this.synthTelemetry(endTime - startTime, stacks, synthOrigin);
+
+    if (stacks.length === 0) {
+      logger.error("ERROR: No Terraform code synthesized.");
+    }
+
+    const stackNames = stacks.map((s) => s.name);
+    const orphanedDirectories = existingDirectories.filter(
+      (e) => !stackNames.includes(path.basename(e))
+    );
+
+    for (const orphanedDirectory of orphanedDirectories) {
+      fs.rmSync(orphanedDirectory, { recursive: true });
+    }
+
+    return stacks;
+  }
+
+  public static async readSynthesizedStacks(
+    outdir: string
+  ): Promise<SynthesizedStack[]> {
+    const manifestPath = path.join(outdir, Manifest.fileName);
+    if (!(await fs.pathExists(manifestPath))) {
+      throw new Error(`Could not find manifest file at ${manifestPath}.`);
```

Comment:
```suggestion
      throw new Error(`Could not find manifest file at ${manifestPath}. In case --skip-synth was passed, please try again without the flag.`);
```

Reply:
Unsure if this is the right place, but overall I wanted to see if there are ways to catch the fact that there is no synthed JSON available on disk, and highlight that the flag may have been passed.